### PR TITLE
Add support for optional object fields

### LINKED ISSUE #13: Support optional fields in object types
```typescript
type Foo = {
  foo?: string;
}
```

Currently this type is not supported because the `foo` field is optional. We need to add support to `FixedObjectType` and its parser to generate Elm `Maybe` types for fields like this.

### PR DESCRIPTION
**Pull request by sentience**
TypeScript object types with optional fields like this:

```typescript
type Foo = {
  always: string;
  sometimes?: string;
}
```

will now generate an Elm type alias with `Maybe` types for the optional fields:

```elm
type alias Foo =
    { always : String
    , sometimes : Maybe String
    }
```

To support this, we must generate a more complex JSON object encoder, which builds up a list of _maybe_ fields, and then uses only the ones that are not `Nothing`:

```elm
fooEncoder : Foo -> Value
fooEncoder foo =
    Encode.object
        ([ foo.always |> (\value -> Just ( "always", Encode.string value ))
         , foo.sometimes |> Maybe.map (\value -> ( "sometimes", Encode.string value ))
         ]
            |> List.filterMap identity
        )
```

Because I was feeling increasingly guilty about making changes to `fixed-object-type/parser` without a test suite to ensure I wasn't breaking things, this also introduces some unit tests, currently covering `fixed-object-type` and its parser. You can run the tests with `npm test`.

Closes #13.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "outDir": "dist",
     "jsx": "preserve",
     "sourceMap": true,
-    "skipLibCheck": true,
```
Comment (sentience) [inline tsconfig.json:19]: Jest was objecting to the trailing comma, which is invalid JSON.

File: src/prop-types/fixed-object-type/parser.ts [hunk 2]
```diff
@@ -3,24 +3,23 @@
...
   | 'beforePrimitiveType'
-  | 'afterField'
   | 'end'
   | 'done';
 
 export default function parser(resolvedType: string) {
   let parsePosition = 0;
   let parserState: ParserState = 'start';
-  let fields: { name: string; type: string }[] = [];
+  let fields: { name: string; type: string; required: boolean }[] = [];
```
Comment (sebpearce) [inline src/prop-types/fixed-object-type/parser.ts:20]: 👍

File: src/prop-types/fixed-object-type/parser.ts [hunk 3]
```diff
@@ -32,16 +31,24 @@ export default function parser(resolvedType: string) {
...
               {
-                regExp: /^(\w+): /,
+                regExp: /^(\w+\??): /,
                 capture: (fieldName) => {
-                  nextFieldName = fieldName;
+                  const optionalSuffix = '?';
+                  nextFieldIsRequired = !fieldName.endsWith(optionalSuffix);
+                  nextFieldName = nextFieldIsRequired
+                    ? fieldName
+                    : fieldName.slice(0, -optionalSuffix.length);
```
Comment (sebpearce) [inline src/prop-types/fixed-object-type/parser.ts:51]: Always enjoy reading code like this where variables are self-documenting (`optionalSuffix`). 👌

**Review by sebpearce — APPROVED** [review]
LGTM!

**Pull request merged**